### PR TITLE
Wait for config volume to be ready

### PR DESCRIPTION
**Pull request by tuliren**
## What
- Previously, based on logs from kube acceptance test, I suspected that the config volume may not always be ready when the server app launches.
- Logs from [this run](https://github.com/airbytehq/airbyte/runs/3109170412) proved this hypothesis:

```
Caused by: java.nio.file.NoSuchFileException: /configs/config
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newDirectoryStream(UnixFileSystemProvider.java:432)
	at java.base/java.nio.file.Files.newDirectoryStream(Files.java:476)
	at java.base/java.nio.file.Files.list(Files.java:3765)
	at io.airbyte.config.persistence.FileSystemConfigPersistence.listDirectories(FileSystemConfigPersistence.java:115)
	at io.airbyte.config.persistence.FileSystemConfigPersistence.dumpConfigs(FileSystemConfigPersistence.java:106)
	at io.airbyte.config.persistence.ValidatingConfigPersistence.dumpConfigs(ValidatingConfigPersistence.java:83)
	at io.airbyte.config.persistence.DatabaseConfigPersistence.lambda$loadData$3(DatabaseConfigPersistence.java:104)
```

## How
- The solution is to wait for the config volume to be ready before constructing the file system config persistence.
- Logs from [this run](https://github.com/airbytehq/airbyte/runs/3110659452) proved that this approach worked:

```
2021-07-20 04:38:55 INFO i.a.c.p.FileSystemConfigPersistence(createWithValidation):62 - ***workspace_app_root=/workspace/server/logs*** - Constructing file system config persistence (root: /configs)
2021-07-20 04:38:55 WARN i.a.c.p.FileSystemConfigPersistence(createWithValidation):67 - ***workspace_app_root=/workspace/server/logs*** - Config volume is not ready yet (waiting time: 0 s)
2021-07-20 04:38:58 INFO i.a.c.p.FileSystemConfigPersistence(createWithValidation):71 - ***workspace_app_root=/workspace/server/logs*** - Config volume is ready (waiting time: 3 s)
```

### CODE REVIEW

File: airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/FileSystemConfigPersistence.java [hunk 1]
```diff
@@ -57,14 +57,13 @@
...
-  public static ConfigPersistence createWithValidation(final Path storageRoot) throws IOException {
+  public static ConfigPersistence createWithValidation(final Path storageRoot) {
     return new ValidatingConfigPersistence(new FileSystemConfigPersistence(storageRoot));
   }
 
-  public FileSystemConfigPersistence(final Path storageRoot) throws IOException {
+  public FileSystemConfigPersistence(final Path storageRoot) {
     this.storageRoot = storageRoot;
     this.configRoot = storageRoot.resolve(CONFIG_DIR);
-    Files.createDirectories(configRoot);
```
Comment (tuliren) [inline airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/FileSystemConfigPersistence.java:67]: This was added by me in #4670. But I should not have done that. This statement covered the corner case that sometimes the config volume is not ready when server app starts, at least in kube acceptance test.

File: airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/ConfigPersistenceBuilderTest.java [hunk 2]
```diff
@@ -155,20 +155,6 @@ public void testCreateDbPersistenceWithoutSetupDatabase() throws Exception {
         dbPersistence.dumpConfigs());
   }
 
-  @Test
-  public void testCreateFileSystemConfigPersistence() throws Exception {
```
Comment (davinchia) [inline airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/ConfigPersistenceBuilderTest.java:159]: why delete this test?

**Review by davinchia — APPROVED** [review]

File: airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/ConfigPersistenceBuilderTest.java (hunk 2 above)
Comment (tuliren) [reply airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/ConfigPersistenceBuilderTest.java:159]: The `ConfigPersistenceBuilder#getFileSystemPersistence` method is redundant. So this method is deleted, together with this test.

Comment (davinchia) [reply airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/ConfigPersistenceBuilderTest.java:159]: makes sense!

**Pull request merged**